src/AWS.Deploy.Orchestration/Data/AWSResourceQueryer.cs

```diff
@@ -438,10 +440,56 @@ public async Task<List<Stack>> GetCloudFormationStacks()
             return await cloudFormationClient.Paginators.DescribeStacks(new DescribeStacksRequest()).Stacks.ToListAsync();
         }
 
-        public async Task<GetCallerIdentityResponse> GetCallerIdentity()
+        public async Task<GetCallerIdentityResponse> GetCallerIdentity(AWSCredentials awsCredentials, string awsRegion)
         {
-            using var stsClient = _awsClientFactory.GetAWSClient<IAmazonSecurityTokenService>();
-            return await stsClient.GetCallerIdentityAsync(new GetCallerIdentityRequest());
+            var request = new GetCallerIdentityRequest();
+
+            try
+            {
+                _awsClientFactory.ConfigureAWSOptions(awsOption =>
+                {
+                    awsOption.Credentials = awsCredentials;
+                    awsOption.Region = RegionEndpoint.GetBySystemName(awsRegion);
+                });
+
+                using var stsClient = _awsClientFactory.GetAWSClient<IAmazonSecurityTokenService>();
+                return await stsClient.GetCallerIdentityAsync(request);
+            }
+            catch (Exception ex)
+            {
+                var regionEndpoint = RegionEndpoint.GetBySystemName(awsRegion);
```

Comment:
I believe this throws if awsRegion is not a value that the SDK knows about (for example, a newer region), and you're in a catch block. Is that handled as expected?

Reply:
I checked this and it does not throw an exception. It returns an endpoint with an "Unknown" name.

Reply:
To be clear the display name will be `Unknown` and the system name will be what you pass in for `awsRegion`. It was the classes way of dealing with regions not defined in the `endpoints.json` file.